### PR TITLE
#1206 Fix Catalyst optimizer workaround bug.

### PR DESCRIPTION
**Pull request by yruslan**

### CODE REVIEW

File: spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/OptimizerTimeTracker.scala [hunk 1]
```diff
@@ -116,11 +117,23 @@ class OptimizerTimeTracker(inputDf: DataFrame, isWorkaroundEnabled: Boolean)(imp
...
     t1 - t0
   }
 
+  /**
+   * Returns the optimized execution plan of a dataframe. Ensures the optimizer is invoked.
+   * `df.queryExecution.optimized` can return a cached of an already calculated value.
+   * This method ensures that the optimizer actually runs.
+   *
+   * @param df A dataframe to calculate execution plan optimization time
+   * @return Elapsed time in milliseconds
```
Comment (dk1844) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/OptimizerTimeTracker.scala:131]: The description of the returned value seems incorrect.

File: spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/OptimizerTimeTracker.scala [hunk 2]
```diff
@@ -26,7 +26,7 @@ class OptimizerTimeTracker(inputDf: DataFrame, isWorkaroundEnabled: Boolean)(imp
   private val log = LoggerFactory.getLogger(this.getClass)
 
   private val maxToleratedPlanGenerationPerRuleMs = 100L
-  private val initialElapsedTimeBaselineMs = 300L
+  private val initialElapsedTimeBaselineMs = 250L
```
Comment (benedeki) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/OptimizerTimeTracker.scala:29]: Just curious why the change? 😉

Comment (yruslan) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/OptimizerTimeTracker.scala:29]: It just while observing the behavior of more jobs I saw that 300  ms initial baseline is a little too high.

**Comment by yruslan** [comment]
Btw, I'm still not going to merge this. @kevinwallimann, said the streaming conformance stuck again. So I need to confirm whether it is related to the Catalyst optimizer bug or it is something else.

**Comment by yruslan** [comment]
Guys, please, approve again.

This version is not perfect, but it works for both batch and streaming.
We'll continue on making this workaround better, but at least it will work in `2.5.0`.

**Comment by yruslan** [comment]
👍  Thanks. I'm going to rebase since I might return to peek on the intermediate steps in the future.
